lumicks/pylake/channel.py

```diff
@@ -583,8 +591,17 @@ def stop(self):
             raise IndexError("End of empty time series is undefined")
 
     @property
-    def _timesteps(self):
-        return np.unique(np.diff(self.timestamps)).astype(int)
+    def sample_rate(self) -> Union[float, None]:
+        """The data frequency or `None` if it is variable"""
+        timestep = self._timesteps
+        if len(timestep) > 1:
+            return None
+        else:
+            return 1e9 / timestep[0]  # Hz
+
+    @property
+    def _timesteps(self) -> npt.ArrayLike:
+        return np.unique(np.diff(self.timestamps)).astype(int)  # ns
```

Comment:
Why do you cast this to an `int` explicitly?

There's a risk associated with this for `TimeSeries`, because the precision may be insufficient to capture the timestep. If you downsample to 0.1 Hz for instance, you'll have a timestamp difference of `1e10` which when cast like this will overflow and therefore wrap around. Also note that timestamps are typically not regular `dtype=int` in Pylake, but `np.int64` because of the precision issues.

I'd omit the `astype(...)` if there's no specific reason to have it. If it's needed for downstream stuff, `np.int64` would be a better choice.

Reply:
I removed the type conversion, which was a left over from old code. Also I ensured that property `_timesteps` in `Contiguous` returns an array of dtype `np.int64`.